File: src/main/java/com/codurance/module1/TestDoubles/Stub/OrderProcessor.java
package com.codurance.module1.TestDoubles.Stub;

public class OrderProcessor {
    private PaymentGateway paymentGateway;

    public OrderProcessor(PaymentGateway paymentGateway) {
        this.paymentGateway = paymentGateway;
    }

    public void processOrder(double amount) throws PaymentChargeFailed {
        if (!paymentGateway.charge(amount)) {
            throw PaymentChargeFailed.with(amount);
        }
    }
}

File: typescript-examples/src/module1/testDoubles/stub/OrderProcessor.ts
import {PaymentGateway} from "./PaymentGateway.ts";
import {PaymentChargeFailed} from "./PaymentChargeFailed.ts";

export class OrderProcessor {
    private readonly paymentGateway: PaymentGateway;

    constructor(paymentGateway: PaymentGateway) {
        this.paymentGateway = paymentGateway;
    }

    public processOrder(amount: number): void {
        if (!this.paymentGateway.charge(amount)) {
            throw PaymentChargeFailed.with(amount);
        }
    }
}